src/Bridges/Constraint/indicator_sos.jl

```diff
@@ -0,0 +1,79 @@
+"""
+    IndicatorSOS1Bridge{T, BC <: MOI.AbstractScalarSet}
+
+The `IndicatorSOS1Bridge` replaces an indicator constraint of the following form:
+``z \\in \\mathbb{B}, z == 1 \\implies f(x) \\leq b`` with a SOS1 constraint:
+``z \\in \\mathbb{B}, w \\leq 0, f(x) + w \\leq b, SOS1(w, z)``.
+`GreaterThan` constraints are handled in a symmetric way:
+``z \\in \\mathbb{B}, z == 1 \\implies f(x) \\geq b`` is reformulated as:
+``z \\in \\mathbb{B}, w \\geq 0, f(x) + w \\geq b, SOS1(w, z)``.
+Other scalar sets are handled without a bound constraint:
+``z \\in \\mathbb{B}, z == 1 \\implies f(x) == b`` is reformulated as:
+``z \\in \\mathbb{B}, w \\text{ free}, f(x) + w == b, SOS1(w, z)``.
+
+If `BC !<: Union{LessThan, GreaterThan}`, `bound_constraint_index` is `nothing`.
+"""
+struct IndicatorSOS1Bridge{T, BC <: MOI.AbstractScalarSet, MaybeBC <: Union{MOI.ConstraintIndex{MOI.SingleVariable, BC}, Nothing}} <: AbstractBridge
+    w_variable_index::MOI.VariableIndex
+    bound_constraint_index::MaybeBC
+    sos_constraint_index::MOI.ConstraintIndex{MOI.VectorOfVariables, MOI.SOS1{T}}
+    linear_constraint_index::MOI.ConstraintIndex{MOI.ScalarAffineFunction{T}, BC}
+end
+
+function bridge_constraint(::Type{IndicatorSOS1Bridge{T,BC,MaybeBC}}, model::MOI.ModelLike, f::MOI.VectorAffineFunction{T}, s::MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE, BC}) where {T <: Real, BC, MaybeBC}
+    f_scalars = MOIU.eachscalar(f)
+    (w, bound_constraint) = _add_bound_constraint!(model, BC)
+    z = convert(MOI.SingleVariable, f_scalars[1]).variable
+    sos_vector = MOI.VectorOfVariables([w, z])
+    sos_constraint = MOI.add_constraint(model, sos_vector, MOI.SOS1{T}([0.4, 0.6]))
+    affine_func = f_scalars[2]
+    MOIU.operate!(+, T, affine_func, MOI.SingleVariable(w))
```

Comment:
If `f` is `VectorOfVariables`, `affine_func` is `SingleVariable` and it cannot hold the sum so to be safe, do `some_name = MOIU.operate!(...`

Reply:
done